source/protobuf/session.proto

```diff
@@ -54,6 +62,24 @@ message EnumerateDevicesResponse {
   repeated DeviceProperties devices = 1;
 }
 
+message SoftwareProperties {
+  string package_id = 1;
+  string package_version = 2;
+  string product_name = 3;
+}
+
+message EnumerateInstalledSoftwareRequest {}
+
+message EnumerateInstalledSoftwareResponse {
+  repeated SoftwareProperties software = 1;
+}
+
+message EnumerateInstalledSoftwareIncludeHiddenPackagesRequest {}
```

Comment:
It feels a little weird to make this 2 separate methods instead of passing an argument. 
@harsha-bhushan thoughts on how this should be exposed?

Reply:
This is mostly just me being a newb to proto files. I agree that an additional argument would be better - let me revisit this.

Reply:
I changed the Request to take an argument (and tested everything out again). Let me know if this looks better.

Reply:
The single method with argument looks good.